core/src/test/java/com/linecorp/armeria/client/AggregatedResponseAsTest.java

```diff
@@ -104,6 +220,83 @@ void jsonObject_customMapper() {
         assertThat(entity.content()).isEqualTo(myObject);
     }
 
+    @Test
+    void jsonObject_customMapper_withHttpStatusPredicate_status200() {
+        final AggregatedHttpResponse response =
+                AggregatedHttpResponse.of(headers, HttpData.ofUtf8("{ 'id': 10 }"));
+
+        final JsonMapper mapper = JsonMapper.builder()
+                                            .enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
+                                            .build();
+        final ResponseEntity<MyObject> entity = AggregatedResponseAs.json(
+                MyObject.class, mapper, HttpStatusPredicate.of(HttpStatus.OK)).as(response);
+        final MyObject myObject = new MyObject();
+        myObject.setId(10);
+        assertThat(entity.content()).isEqualTo(myObject);
+    }
+
+    @Test
+    void jsonObject_customMapper_withHttpStatusPredicate_status500() {
+        final AggregatedHttpResponse response =
+                AggregatedHttpResponse.of(headersStatusCode500, HttpData.ofUtf8("{ 'id': 10 }"));
+
+        final JsonMapper mapper = JsonMapper.builder()
+                                            .enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
+                                            .build();
+        final ResponseEntity<MyObject> entity = AggregatedResponseAs.json(
+                MyObject.class, mapper, HttpStatusPredicate.of(HttpStatus.INTERNAL_SERVER_ERROR)).as(response);
+        final MyObject myObject = new MyObject();
+        myObject.setId(10);
+        assertThat(entity.content()).isEqualTo(myObject);
+    }
```

Comment:
The two tests are the same except for the HTTP status. Should we consolidate the two tests using `@ParameterizedTest()`?